hmcts/pre-api
# S28 3263: Accept Terms and Conditions

### JIRA ticket(s)
- S28-3260
- S28-3263


### Change description
- Add terms and conditions acceptance field to `UserDTO` 
- Add terms and conditions acceptance endpoint `POST /accept-terms-and-conditions/{termsId}`
- Add some methods for supporting testing




> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**
- Test 1: 
    - Create a user
    - Get the user's details `GET /user/{id}` and see new field `terms_accepted`
    - See two fields `APP` and` PORTAL` both with `false` values 
- Test 2 (do this for both terms types: APP/PORTAL): 
    - Create a new app terms and conditions via `POST /testing-support/create-terms-and-conditions/{termsType}`
    - Call accept terms and conditions `POST /accept-terms-and-conditions/{termsId}`
    - Get user's details `GET /user/{id}` and see `terms_accepted.APP/PORTAL` is true
    - Call testing support endpoint to outdate `/testing-support/outdate-all-user-acceptances` to set all user acceptances to 1 year ago (outdating the acceptance)
    -  Get user's details `GET /user/{id}` and see `terms_accepted.APP/PORTAL` is false
- Test 3: 
    - Create a new app terms and conditions via `POST /testing-support/create-terms-and-conditions/{termsType}`
    - Call accept terms and conditions `POST /accept-terms-and-conditions/{termsId}`
    - Get user's details `GET /user/{id}` and see `terms_accepted.APP` is true
    - Create a new app terms and conditions via `POST /testing-support/create-terms-and-conditions/{termsType}`
    - Get user's details `GET /user/{id}` and see `terms_accepted.APP` is false
